python/dgl/graph_index.py

```diff
@@ -677,6 +677,20 @@ def neighbor_sampling(self, seed_ids, expand_factor, num_hops, neighbor_type, no
         return [SubgraphIndex(rst(i), self, utils.toindex(rst(num_subgs + i)),
                               utils.toindex(rst(num_subgs * 2 + i))) for i in range(num_subgs)]
 
+    def random_walk(self, seeds, num_traces, num_hops):
+        """Random walk sampling.
+
+        Returns a user Tensor of random walk traces with shape
+        (num_seeds, num_traces, num_hops + 1)
+        """
+        if len(seeds) == 0:
+            return utils.toindex([])
+
+        seeds = seeds.todgltensor()
+        traces = _CAPI_DGLGraphRandomWalk(self._handle, seeds, num_traces, num_hops)
+
+        return F.zerocopy_from_dlpack(traces.to_dlpack())
+
```

Comment:
Change it to global function.

Reply:
Similarly, we probably need a `Sampler` class that wraps the C++ `Sampler` class?

Reply:
Agree.